components/admin_console/admin_definition.jsx

```diff
@@ -2454,6 +2454,34 @@ const AdminDefinition = {
                             )
                         ),
                     },
+                    {
+                        type: Constants.SettingsTypes.TYPE_BOOL,
+                        key: 'LdapSettings.EnableAdminFilter',
+                        label: t('admin.ldap.enableAdminFilterTitle'),
+                        label_default: 'Enable Admin Filter:',
+                        isDisabled: it.both(
+                            it.stateIsFalse('LdapSettings.Enable'),
+                            it.stateIsFalse('LdapSettings.EnableSync'),
+                        ),
+                    },
+                    {
+                        type: Constants.SettingsTypes.TYPE_TEXT,
+                        key: 'LdapSettings.AdminFilter',
+                        label: t('admin.ldap.adminFilterTitle'),
+                        label_default: 'Admin Filter:',
+                        help_text: t('admin.ldap.adminFilterFilterDesc'),
+                        help_text_default: '(Optional) Enter an AD/LDAP filter to use for designating System Admins. The users selected by the query will have access to your Mattermost server as System Admins. By default, System Admins have complete access to the Mattermost System Console.\n \nExisting members that are identified by this attribute will be promoted from member to System Admin upon next login. The next login is based upon Session lengths set in **System Console > Session Lengths**. It is highly recommend to manually demote users to members in **System Console > User Management** to ensure access is restricted immediately.\n \nNote: If this filter is removed/changed, System Admins that were promoted via this filter will be demoted to members and will not retain access to the System Console. When this filter is not in use, System Admins can be manually promoted/demoted in **System Console > User Management**.',
+                        help_text_markdown: true,
+                        placeholder: t('admin.ldap.adminFilterEx'),
+                        placeholder_default: 'E.g.: "(objectClass=admins)"',
+                        isDisabled: it.either(
+                            it.configIsFalse('LdapSettings', 'EnableAdminFilter'),
```

Comment:
Currently the Admin Filter attribute is not automatically enabled when we set "Enable Admin Filter" to true. We need to confirm the changes by clicking on Save button and only then this field gets enabled. Can we instead just check the state, something like this:
`it.StateIsFalse('LdapSettings.EnableAdminFilter')`